src/main/java/seedu/address/model/person/Person.java

```diff
@@ -132,24 +135,19 @@ public boolean equals(Object other) {
         if (other == this) {
             return true;
         }
-
-        // instanceof handles nulls
         if (!(other instanceof Person)) {
             return false;
         }
-
         Person otherPerson = (Person) other;
         return name.equals(otherPerson.name)
                 && studentClass.equals(otherPerson.studentClass)
                 && phone.equals(otherPerson.phone)
-                && tags.equals(otherPerson.tags)
-                && groups.equals(otherPerson.groups);
```

Comment:
Maybe since we are no longer using this groups field, we can remove it from the Person class to reduce ambiguity?